### PR TITLE
GH-1083 - Adds AmChart element class names & Scatter fix

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @jbrundage @dtsnell4 - Please Review

-Adds AmChart element class names
-Fixes change in scatter label behavior (need to monitor this more closely)

Fixes GH-1083

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/amchart/CommonXY.js [hunk 1]
```diff
@@ -124,10 +124,11 @@
         if (this.xAxisType() === "ordinal") {
             this._chart.valueAxes[0].integersOnly = true;
             this._chart.valueAxes[0].labelFunction = function(val) {
-                if (val > 0) {
-                    return typeof(context.data()[val]) !== "undefined" ? context.data()[val][0] : ""; // shift chart right by 1 so it does not start on axis
-                }
-                return "";
+                return context.data()[--val][0];
+                // if (val > 0) {
```
Comment (mlzummo) [inline src/amchart/CommonXY.js]: leaving this commented code in there on purpose (i think amcharts changed something so i have to do this diff)

**Comment by dtsnell4** [comment]
looks ok to me

**Comment by mlzummo** [comment]
found amcharts bug in the process ... emailing them for a solution

File: src/amchart/CommonXY.js [hunk 2]
```diff
@@ -121,13 +121,13 @@
...
+
         if (this.xAxisType() === "ordinal") {
             this._chart.valueAxes[0].integersOnly = true;
+            this._chart.valueAxes[0].maximum = this.data().length;
             this._chart.valueAxes[0].labelFunction = function(val) {
-                if (val > 0) {
-                    return typeof(context.data()[val]) !== "undefined" ? context.data()[val][0] : ""; // shift chart right by 1 so it does not start on axis
-                }
-                return "";
+                return context.data().length && val > 0 ? context.data()[--val][0] : "";
```
Comment (GordonSmith) [inline src/amchart/CommonXY.js]: Should be just ```val - 1```.

File: test/runner-rawgit.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    </script>
+    <script>
+        mocha.setup('bdd');
+        var assert = chai.assert;
+        require(["./widgets", "./persist"], function () {
+            mocha.run();
+        });
+    </script>
+</body>
+</html>
```
Comment (GordonSmith) [inline test/runner-rawgit.html]: we already have a runner_rawgit.html, what is different?

**Comment by mlzummo** [comment]
@GordonSmith removed file

**Comment by mlzummo** [comment]
@GordonSmith updated val-1

**Pull request merged**